File: easyevent_frontend/easyevent/src/models/dto/UserDTO.ts
export class UserDTO {
    email: string;
    name: string;
    profilePicture: string|null;
    language: string|null;
    oldPassword: string|null;
    newPassword: string|null;
    token: string|null;
    id: number|null;
    active: boolean;
    isLocalPasswordSet: boolean;
    isAdmin: boolean;

    constructor(email: string, name: string, profilePicture: string|null, language: string|null, oldPassword: string|null, newPassword: string|null, token: string|null, id: number|null, active: boolean, isLocalPasswordSet: boolean, isAdmin: boolean) {
        this.email = email;
        this.name = name;
        this.profilePicture = profilePicture;
        this.language = language;
        this.oldPassword = oldPassword;
        this.newPassword = newPassword;
        this.token = token;
        this.id = id;
        this.active = active;
        this.isLocalPasswordSet = isLocalPasswordSet;
        this.isAdmin = isAdmin;
    }
}

export default UserDTO;

File: easyevent_backend/src/main/java/com/benny1611/easyevent/dto/UserDTO.java
package com.benny1611.easyevent.dto;

import com.benny1611.easyevent.util.ValidUserDTOPasswordChange;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
@ValidUserDTOPasswordChange
public class UserDTO {
    @NotBlank
    @Email(message = "Email must be present and valid")
    private String email;
    private String name;
    private String profilePicture;
    private String language;
    private String oldPassword;
    private String newPassword;
    private String token;
    private boolean active;
    private boolean oauthUser;
}
